src/index.js

```diff
@@ -13,45 +13,94 @@
  * @returns {typeof EPromise} EPromise class
  */
 export default function makeEPromiseClass(BasePromise) {
-  // A remoteRelay must additionally have an AWAIT_FAR method
-  const localRelay = {
-    GET(p, key) {
-      return p.then(o => o[key]);
-    },
-    PUT(p, key, val) {
-      return p.then(o => (o[key] = val));
-    },
-    DELETE(p, key) {
-      return p.then(o => delete o[key]);
-    },
-    POST(p, optKey, args) {
-      if (optKey === undefined || optKey === null) {
-        return p.then(o => o(...args));
-      }
-      return p.then(o => o[optKey](...args));
-    },
-  };
-
-  const relayToPromise = new WeakMap();
+  const presenceToResolvedRelay = new WeakMap();
   const promiseToRelay = new WeakMap();
 
+  // This special relay accepts Promises, and forwards
+  // the resolved Presence to its corresponding resolvedRelay.
+  //
+  // If passed a Promise that is not a Presence, perform
+  // the corresponding local operation.
+  let forwardingRelay;
   function relay(p) {
-    return promiseToRelay.get(p) || localRelay;
+    return promiseToRelay.get(p) || forwardingRelay;
   }
 
   class EPromise extends BasePromise {
-    static makeRemote(remoteRelay) {
-      const promise = this.resolve(remoteRelay.AWAIT_FAR());
-      relayToPromise.set(remoteRelay, promise);
-      promiseToRelay.set(promise, remoteRelay);
-      return promise;
+    static makePresence(executor, unresolvedRelay) {
+      let presenceResolve;
+      let presenceReject;
+      let relayResolve;
+      let relayReject;
+      const presenceP = new EPromise((resolve, reject) => {
+        presenceResolve = resolve;
+        presenceReject = reject;
+      });
+      const relayP = new EPromise((resolve, reject) => {
+        relayResolve = resolve;
+        relayReject = reject;
+      });
+
+      function postpone(forwarded) {
```

Comment:
Can we put the `postpone` function within the `if (!unresolvedRelay) {` ?

Reply:
Fixed in next commit.